src/main/java/com/SAS/User/Privileges.java

```diff
@@ -1,13 +1,76 @@
 /**
- * The class represents the privileges of registered user
+ * The class represents the a privileges singleton that will be held by the role class
  */
 
 package com.SAS.User;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
 
 public class Privileges {
 
-    //Will be implemented
-    public void edit(String privilege) throws Exception {
-        throw new Exception("Not implemented");
+    private volatile static Privileges instance = null;
+    public HashMap<String, HashSet<String>> privileges;
+
+    /**
+     * constructor for the singleton
+     */
+    private Privileges(){
+        privileges = new HashMap<>();
+        initPrivileges();
+    }
+
+    public static Privileges getInstance() {
+        if(instance == null){
+            synchronized(Privileges.class){
+                if(instance == null) {
```

Comment:
Why it's necessary to double privileges for null?

Reply:
So there will be only one instance of Privileges, otherwise there may be 2 threads that will enter and create the Privileges  object.